kafka-client/src/test/java/dev/responsive/kafka/integration/ResponsiveSessionStoreIntegrationTest.java

```diff
@@ -0,0 +1,261 @@
+        .to(outputTopic());
+
+    // When:
+    final Map<String, Object> properties = getMutablePropertiesWithStringSerdes();
+    properties.put(STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+    final KafkaProducer<String, String> producer = new KafkaProducer<>(properties);
+    try (
+        final ResponsiveKafkaStreams kafkaStreams =
+            new ResponsiveKafkaStreams(builder.build(), properties);
+    ) {
+      startAppAndAwaitRunning(Duration.ofSeconds(15), kafkaStreams);
+      pipeRecords(producer, inputTopic(), inputEvents);
+
+      assertThat(
+          outputLatch.await(20_000, TimeUnit.MILLISECONDS),
+          Matchers.equalTo(true)
+      );
+
+      assertThat(actualPeeks, Matchers.hasSize(expectedPeeks.size()));
+      for (var i = 0; i < actualPeeks.size(); i++) {
+        var actual = actualPeeks.get(i);
+        var expected = expectedPeeks.get(i);
+        assertThat(actual.key, Matchers.equalTo(expected.key));
+        assertThat(actual.value, Matchers.equalTo(expected.value));
+      }
+    }
+  }
+
+  private Windowed<String> windowedKey(final String key, final long startMs, final long endMs) {
+    return new Windowed<>(key, new SessionWindow(startMs, endMs));
+  }
+
+  private Map<String, Object> getMutablePropertiesWithStringSerdes() {
+    final Map<String, Object> properties = getMutableProperties();
+    properties.put(KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+    properties.put(VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+    properties.put(KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+    properties.put(VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+    properties.put(DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.StringSerde.class.getName());
+    properties.put(DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.StringSerde.class.getName());
+    return properties;
+  }
+
+  private Map<String, Object> getMutableProperties() {
+    final Map<String, Object> properties = new HashMap<>(responsiveProps);
+
+    properties.put(KEY_SERIALIZER_CLASS_CONFIG, LongSerializer.class);
+    properties.put(VALUE_SERIALIZER_CLASS_CONFIG, LongSerializer.class);
+    properties.put(KEY_DESERIALIZER_CLASS_CONFIG, LongDeserializer.class);
+    properties.put(VALUE_DESERIALIZER_CLASS_CONFIG, LongDeserializer.class);
+
+    properties.put(APPLICATION_ID_CONFIG, name);
+    properties.put(DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.LongSerde.class.getName());
+    properties.put(DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.LongSerde.class.getName());
+    properties.put(NUM_STREAM_THREADS_CONFIG, 1);
+    properties.put(COMMIT_INTERVAL_MS_CONFIG, 1); // commit as often as possible
+
+    properties.put(consumerPrefix(REQUEST_TIMEOUT_MS_CONFIG), 5_000);
+    properties.put(consumerPrefix(SESSION_TIMEOUT_MS_CONFIG), 5_000 - 1);
+
+    properties.put(consumerPrefix(MAX_POLL_RECORDS_CONFIG), 1);
+
+    properties.put(ResponsiveConfig.STORE_FLUSH_RECORDS_TRIGGER_CONFIG, 1);
```

Comment:
This is the config I was referring to above, just fyi -- worth running just once with this disabled